apphub/image_generation/dcgan/dcgan_tensorflow.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""DCGAN example using MNIST data set."""
+import tensorflow as tf
+from tensorflow.python.keras import layers
+
+import fastestimator as fe
+from fastestimator.backend import cross_entropy
+from fastestimator.dataset import mnist
+from fastestimator.op import TensorOp
+from fastestimator.op.numpyop import ExpandDims, Normalize
+from fastestimator.op.tensorop.model import ModelOp, UpdateOp
+
+
+class GLoss(TensorOp):
+    def forward(self, data, state):
+        return cross_entropy(y_pred=data, y_true=tf.ones_like(data), from_logits=True)
+
+
+class DLoss(TensorOp):
+    def forward(self, data, state):
+        true_score, fake_score = data
+        real_loss = cross_entropy(y_pred=true_score, y_true=tf.ones_like(true_score), from_logits=True)
+        fake_loss = cross_entropy(y_pred=fake_score, y_true=tf.zeros_like(fake_score), from_logits=True)
+        total_loss = real_loss + fake_loss
+        return total_loss
+
+
+def generator():
```

Comment:
do we want dcgan architectures to be in the architecture folder?

Reply:
nope, it's not a fixed architecture like lenet.